### ISSUE TITLE
[App]: Cannot see the plot of metrics logged in a resumed run

### ISSUE THREAD

**Issue by najwalb** [wandb/wandb#6589]
### Current Behavior

I cannot see the metrics I log when resuming a run in the wandb interface section. This mostly happens with my old runs. I tried creating a new public project to showcase this issue but was not able to reproduce it. The screenshot below is from my private run.
<img width="733" alt="Screenshot 2023-11-10 at 18 36 40" src="https://github.com/wandb/wandb/assets/107994360/4c9750a8-f368-4792-9eea-ce442519d209">

### Expected Behavior

New metrics are plotted in their corresponding section in the resumed wandb runs, as in the new public run below:
<img width="749" alt="Screenshot 2023-11-10 at 18 42 20" src="https://github.com/wandb/wandb/assets/107994360/b4cdb00c-fb8d-4cdc-9372-865c08421b40">

### Steps To Reproduce

(create a new conda env and install wandb 0.16.0)

I wrote this script to reporduce the issue but it only happens with my old run (the private one). It happens with most of my private old runs in fact.

'''
A script to show how wandb fails to log new metrics to some old runs. 
the run in debugging-wandb works fine, the one in retrodiffuser does not. 
So far did not manage to reproduce this in a public wandb project.
'''

```
import wandb
import random

# start a new wandb run to track this script
# 1. The run id, etc.
# data from dummy run
# run_id = "qet1ecmx" 
# project = "Debugging-Wandb"
# data for my own (private) run
run_id = "bxwydbhx" 
project = "retrodiffuser"

entity = "najwalb"
resume_nb = 5

# run this to create a dummy run for the first time
# wandb.init(
#     # set the wandb project where this run will be logged
#     project=project, 
#     # track hyperparameters and run metadata
#     config={
#         "learning_rate": 0.02,
#         "architecture": "CNN",
#         "dataset": "CIFAR-100",
#         "epochs": 10,
#     }
# )

def get_run_path(run_id, project, entity):
    return f"{entity}/{project}/{run_id}"

def get_run(run_path):
    api = wandb.Api()
    run = api.run(run_path)
    return run

run_path = get_run_path(run_id, project, entity)
run = get_run(run_path)

# simulate training
step = run.lastHistoryStep+1
with wandb.init(id=run_id, project=project, entity=entity, resume="allow") as run:
    epochs = 10
    offset = random.random() / 5
    for epoch in range(2, epochs):
        acc = 1 - 2 ** -epoch - random.random() / epoch - offset
        loss = 2 ** -epoch + random.random() / epoch + offset
        
        # log metrics to wandb
        wandb.log({f"resume_{resume_nb}/acc": acc, f"resume_{resume_nb}/loss": loss}, step=step)
        step += 1
        
# [optional] finish the wandb run, necessary in notebooks
wandb.finish()
```

### Environment

OS: mac sonoma 14.01

Browsers: chrome/mozilla

Version: 0.16.0

### Additional Context

I found one related issue [here](https://github.com/wandb/wandb/issues/1022), but I tried the proposed solution (setting the x axis to Step) and it did not help.

**Comment by umakrishnaswamy** [comment]
hey @najwalb - just to confirm my understanding: you're only running into this issue when trying to resume older runs? would it be possible to provide a link to any/some of those runs so I could take a look on my end?

**Comment by najwalb** [comment]
Hi, yes I can share links to the runs but their private. I could not replicate the issue in a new public project. Here is an example of a run: https://wandb.ai/najwalb/retrodiffuser/runs/95p37rjt?workspace=user-najwalaabid

**Comment by umakrishnaswamy** [comment]
hey @najwalb - I haven't been able to reproduce the issue again so I have a few follow up questions in the meantime:
- are all your older runs that you're trying to resume (but are reaching the faulty behavior) on the same SDK version? (0.16.0)
- do you reach the same behavior when setting resume='must' in `wandb.init()`? (along with the run_id)
- what happens when you log a new metric to a resumed run? (therefore creating a new panel)

**Comment by sydholl** [comment]
WandB Internal User commented: 
umakrishnaswamy commented: 
hey @najwalb - I haven't been able to reproduce the issue again so I have a few follow up questions in the meantime:
- are all your older runs that you're trying to resume (but are reaching the faulty behavior) on the same SDK version? (0.16.0)
- do you reach the same behavior when setting resume='must' in `wandb.init()`? (along with the run_id)
- what happens when you log a new metric to a resumed run? (therefore creating a new panel)

**Comment by najwalb** [comment]
Hi!

- No my older runs are on the previous SDK
- Will try the 'must' keyword now and let you know!
- My problem is with creating a new panel. The screenshots from my first message are from their own panel 'resume_5'. 

Note that I was able to add metrics to these same runs for a while, but since I posted about this issue it became impossible. As a current work around, I changed my pipeline so that I do extra evaluation in separate runs and link all runs from the same experiment with tags and groups.
